MCP server build fails with 'Could not resolve zod' error

## Description

The MCP server build fails with `Could not resolve "zod"` error when following the build instructions.

## Steps to Reproduce

```bash
cd src/mcp
npm install
npm run build
```

## Expected Behavior

Build completes successfully.

## Actual Behavior

Build fails with:
```
✘ [ERROR] Could not resolve "zod"

    lib/checkpoint-schema.js:28:22:
      28 │ const zod_1 = require("zod");
```

## Root Cause

The `build.mjs` sets `absWorkingDir: join(__dirname, '..')` (the `src` directory), causing esbuild to look for `node_modules` in the parent directory instead of `src/mcp/node_modules`.

## Workaround

Run `npm install` in the root directory before building the MCP server:

```bash
npm install  # in root directory
cd src/mcp
npm install
npm run build
```

## Suggested Fix

Either:
1. Update documentation to include `npm install` in root directory
2. Or configure esbuild's `nodePaths` option to include `src/mcp/node_modules`

## Additional Notes

There are also warnings about CommonJS exports in ESM modules (`lib/*.js` files use `exports` but parent `package.json` has `"type": "module"`). These are warnings only and don't block the build.

--- a/src/mcp/build.mjs
+++ b/src/mcp/build.mjs
@@ -29,6 +29,8 @@ try {
     external: [],
     // Allow importing from parent lib directory
     absWorkingDir: join(__dirname, '..'),
+    // Include MCP's node_modules for dependency resolution
+    nodePaths: [join(__dirname, 'node_modules')],
     // No banner needed - we'll add shebang separately
     // Source maps for debugging
     sourcemap: true,

--- a/src/mcp/tools/python-repl.ts
+++ b/src/mcp/tools/python-repl.ts
@@ -511,8 +511,35 @@ async function appendCodeCellToNotebook(
 // =============================================================================
 
 function getBridgePath(): string {
-  // Navigate from src/mcp/tools to src/bridge
-  return path.join(__dirname, "..", "..", "bridge", "gyoshu_bridge.py");
+  // 1. Environment variable override
+  if (process.env.GYOSHU_BRIDGE_PATH) {
+    if (fs.existsSync(process.env.GYOSHU_BRIDGE_PATH)) {
+      return process.env.GYOSHU_BRIDGE_PATH;
+    }
+  }
+
+  // 2. Try multiple fallback paths for different installation scenarios
+  const candidates = [
+    // Original: from src/mcp/tools to src/bridge (development)
+    path.join(__dirname, "..", "..", "bridge", "gyoshu_bridge.py"),
+    // Bundled: from build/ to src/bridge (when running bundled from src/mcp/build/)
+    path.join(__dirname, "..", "bridge", "gyoshu_bridge.py"),
+    // From project root (when cwd is project root)
+    path.join(process.cwd(), "src", "bridge", "gyoshu_bridge.py"),
+    // Alternative: bridge at cwd/bridge
+    path.join(process.cwd(), "bridge", "gyoshu_bridge.py"),
+  ];
+
+  for (const candidate of candidates) {
+    if (fs.existsSync(candidate)) {
+      return candidate;
+    }
+  }
+
+  throw new Error(
+    `Could not find gyoshu_bridge.py. Searched paths:\n${candidates.map(p => `  - ${p}`).join('\n')}\n` +
+    `Set GYOSHU_BRIDGE_PATH environment variable to the correct path.`
+  );
 }
 
 function getBridgeMetaPath(sessionId: string): string {

